base/abstractarraymath.jl

```diff
@@ -93,6 +93,87 @@ function _dropdims(A::AbstractArray, dims::Dims)
 end
 _dropdims(A::AbstractArray, dim::Integer) = _dropdims(A, (Int(dim),))
 
+
+"""
+    insertdims(A; dims)
+
+Return an array with the same data as `A`, but with singleton dimensions specified by
+`dims` inserted.
+The dimensions of `A` and `dims` must be contiguous.
```

Comment:
Of `A` and `dims` together.
`insertdims([1,2,3], dims=(3,))` would be invalid because it jumps over the second dimension.

Reply:
I do think this could be worded slightly better and could help address the other point simultaneously. Maybe:

```suggestion

The new singleton dimensions are inserted immediately before the specified `dims`. The `dims` may repeat, all must be in `1:ndims(A)`, and they must be in sorted order.
```